Disable nw-based monitoring of interface by default

youennf opened the pull request:
#### facc836f8c3e67d3fdde0cca9170428203d21abd
<pre>
Disable nw-based monitoring of interface by default
<a href="https://bugs.webkit.org/show_bug.cgi?id=294114">https://bugs.webkit.org/show_bug.cgi?id=294114</a>
<a href="https://rdar.apple.com/152271504">rdar://152271504</a>

Reviewed by Chris Dumez.

Add a feature flag to enable or disable nw-based monitoring of interfaces.
Update the compilation directives to compile the timer based support of interface monitoring and use this when the flag is disabled.
We put the feature flag as false by default since it triggered a regression on VisionOS.

We do some additional refactoring:
- We recreate the nw_monitoring everytime we restart the interface monitoring
- We add some more logging in NetworkManagerWrapper.
- We try stopping the interface monitoring if a NetworkRTCMonitor is destroyed without being stopped.
- We update NetworkRTCMonitor::m_isStarted to be only available in ASSERT_ENABLED since it is only used for asserts.
- We rename NetworkManager to NetworkRTCSharedMonitor as it better captures what this is.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp:
(WebKit::createNWPathMonitor):
(WebKit::NetworkManager::NetworkManager):
(WebKit::NetworkManager::addListener):
(WebKit::NetworkManager::removeListener):
(WebKit::NetworkManager::adapterTypeFromInterfaceName const):
(WebKit::NetworkManager::updateNetworksFromPath):
(WebKit::NetworkRTCMonitor::~NetworkRTCMonitor):
(WebKit::NetworkRTCMonitor::startUpdatingIfNeeded):
(WebKit::NetworkRTCMonitor::stopUpdating):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::webRTCInterfaceMonitoringViaNWEnabled const):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:

Canonical link: <a href="https://commits.webkit.org/295942@main">https://commits.webkit.org/295942@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5cb76d2d0a1e08330340134c1be0078fc8cb95b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106595 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26346 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16743 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111800 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57191 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27016 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34848 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80946 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109599 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21427 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96186 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61282 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20880 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14293 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56635 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/99235 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90766 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14323 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114664 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/105213 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33733 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24871 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90017 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34097 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92417 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89726 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22909 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34638 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12450 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29359 "Hash f5cb76d2 for PR 46417 does not build (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33658 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39071 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/129524 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33404 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35267 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36757 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35003 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
```diff
@@ -235,6 +235,12 @@ void NetworkRTCProvider::stopResolver(LibWebRTCResolverIdentifier identifier)
+bool NetworkRTCProvider::webRTCInterfaceMonitoringViaNWEnabled() const
+{
+    RefPtr connection = m_connection.get();
+    return connection ? connection->webRTCInterfaceMonitoringViaNWEnabled() : false;
```
`return connection && connection->webRTCInterfaceMonitoringViaNWEnabled();`

youennf replied:
Done

youennf committed:
Disable nw-based monitoring of interface by default
https://bugs.webkit.org/show_bug.cgi?id=294114
rdar://152271504

Reviewed by Chris Dumez.

Add a feature flag to enable or disable nw-based monitoring of interfaces.
Update the compilation directives to compile the timer based support of interface monitoring and use this when the flag is disabled.
We put the feature flag as false by default since it triggered a regression on VisionOS.

We do some additional refactoring:
- We recreate the nw_monitoring everytime we restart the interface monitoring
- We add some more logging in NetworkManagerWrapper.
- We try stopping the interface monitoring if a NetworkRTCMonitor is destroyed without being stopped.
- We update NetworkRTCMonitor::m_isStarted to be only available in ASSERT_ENABLED since it is only used for asserts.
- We rename NetworkManager to NetworkRTCSharedMonitor as it better captures what this is.

* Source/ThirdParty/libwebrtc/Configurations/libwebrtc.exp:
* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp:
(WebKit::createNWPathMonitor):
(WebKit::NetworkManager::NetworkManager):
(WebKit::NetworkManager::addListener):
(WebKit::NetworkManager::removeListener):
(WebKit::NetworkManager::adapterTypeFromInterfaceName const):
(WebKit::NetworkManager::updateNetworksFromPath):
(WebKit::NetworkRTCMonitor::~NetworkRTCMonitor):
(WebKit::NetworkRTCMonitor::startUpdatingIfNeeded):
(WebKit::NetworkRTCMonitor::stopUpdating):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
(WebKit::NetworkRTCProvider::webRTCInterfaceMonitoringViaNWEnabled const):
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:

Canonical link: https://commits.webkit.org/295942@main
